src/OrbitBase/TypedefTest.cpp

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gtest/gtest.h>
+
+#include <utility>
+#include <vector>
+
+#include "OrbitBase/Typedef.h"
+
+namespace {
+struct TestTag {};
+
+struct AddStruct {
+  int operator()(int i, int j) { return i + j; }
+
+  static int Add(int i, const int& j) { return i + j; }
+};
+
+}  // namespace
+
+static int Add(int i, int j) { return i + j; }
+
+namespace orbit_base {
+template <typename T>
+using Wrapper = Typedef<TestTag, T>;
+
+TEST(TypedefTest, CanInstantiate) {
```

Comment:
You should also add a test with a move-only-type (std::unique_ptr<int>) and something that's neither copyable nor movable (I think std::mutex should fit that)